### PR TITLE
Update kubernetes images and config

### PR DESCRIPTION
**Pull request by jlaxson**

### CODE REVIEW

File: kubernetes/config/frontend.yaml [hunk 1]
```diff
@@ -14,7 +13,10 @@ data:
         '': { endpoint: { address: 'scheduler:8982' } },
       },
 
+      allowAcUpdatesForInstanceNamePrefixes: [''],
```
Comment (EdSchouten) [inline kubernetes/config/frontend.yaml:16]: I think it's good to document that it may be useful to set this option, but I don't think we should have it enabled by default. There are many cases in which people want to do pure remote execution, not allowing clients to poison the Action Cache of the cluster.

File: kubernetes/storage.yaml [hunk 2]
```diff
@@ -70,11 +82,16 @@ metadata:
     prometheus.io/scrape: "true"
   name: storage
   namespace: buildbarn
+  labels:
```
Comment (EdSchouten) [inline kubernetes/storage.yaml:85]: What's the purpose of this change and the one below? The same holds for the changes to frontend.yaml.

**Review by EdSchouten — COMMENTED** [review]
Thanks for submitting this!

Could you also add the `setUmask` option to the worker configuration? Thanks!

File: kubernetes/browser.yaml [hunk 3]
```diff
@@ -25,6 +26,10 @@ spec:
         - mountPath: /config/
           name: configs
           readOnly: true
+        resources:
+          requests:
+            cpu: 1
```
Comment (EdSchouten) [inline kubernetes/browser.yaml:31]: That's a lot, right?

I don't think we provide any resource requests/limits right now, because we have no idea at what scale people are going to use clusters like these.

File: kubernetes/config/common.yaml [hunk 4]
```diff
@@ -36,10 +36,26 @@ data:
...
       maximumMessageSizeBytes: 16 * 1024 * 1024,
-      httpListenAddress: ':80'
+      httpListenAddress: ':80',
+      global: {
+        diagnosticsHttpServer: {
+          listenAddress: ':80',
+          enablePrometheus: true,
+          enablePprof: true,
+        },
+        setUmask: { umask: 0 }
```
Comment (EdSchouten) [inline kubernetes/config/common.yaml:57]: My suggestion is to only enable this option in bb_worker.

File: kubernetes/frontend.yaml [hunk 5]
```diff
@@ -12,19 +12,27 @@ spec:
...
       containers:
       - args:
         - /config/frontend.jsonnet
-        image: buildbarn/bb-storage:20200904T173922Z-e3f0e59
+        image: buildbarn/bb-storage:20210312T082612Z-b15cd28
         name: storage
         ports:
         - containerPort: 8980
           protocol: TCP
+        ports:
```
Comment (EdSchouten) [inline kubernetes/frontend.yaml:25]: Shouldn't there be a single `ports` section?

File: kubernetes/frontend.yaml [hunk 6]
```diff
@@ -48,11 +56,16 @@ metadata:
...
   name: frontend
   namespace: buildbarn
+  labels:
+    app: frontend
 spec:
   ports:
   - port: 8980
     protocol: TCP
     name: grpc
+  - port: 80
```
Comment (EdSchouten) [inline kubernetes/frontend.yaml:66]: Why do we want to route traffic to port 80 through a service? This is just for Prometheus to scrape, right?

File: kubernetes/config/storage.yaml [hunk 7]
```diff
@@ -6,26 +6,65 @@ data:
...
         },
         actionCache: {
-          circular: {
-            directory: '/ac',
-            offsetFileSizeBytes: 1024 * 1024,
-            offsetCacheSize: 1000,
-            dataFileSizeBytes: 100 * 1024 * 1024,
-            dataAllocationChunkSizeBytes: 1024 * 1024,
-            instances: ['bb-event-service', 'remote-execution'],
+          completenessChecking: {
```
Comment (joeljeske) [inline kubernetes/config/storage.yaml:37]: There should not be a "completenessChecking" here since there is already one specified at the frontend. This checking is only aware of the CAS of this single shard, and therefor will return some NotFounds for AC items that reference CAS items not allocated to this shard

**Comment by moroten** [comment]
Looks like the current Kubernetes config contains all these updates. Please reopen if I missed something.
